src/util/streamable_generator.h

```diff
@@ -118,6 +131,7 @@ class [[nodiscard]] basic_stream_generator {
   using promise_type = detail::stream_generator_promise<MIN_BUFFER_SIZE>;
 
  private:
+  CompressionMethod _method = CompressionMethod::NONE;
```

Comment:
Either it it is a `_compressionMethod` and the functions are called `setCompressionMethod` etc. or it is a
`_contentEncoding` with functions `setContentEncoding`. I suspect it is the first case,
as this generator does not need to know about http, but simply compresses a stream.